### PR TITLE
Tf20 update to inherit from Keras Layer

### PR DESCRIPTION
**Pull request by psomers3**
I changed the TCN class to inherit from Keras.Layer (and made residual_block a layer as well) so that it is implementable as a typical Keras layer and can utilize the Keras.models save/load features (I added a test script showing this). Everything else of yours should still work (tbh, haven't tested everything though).

I tried to keep the operations as close to how you wrote it as possible.

Awesome repository, btw. Saved me tons of work!

### CODE REVIEW

**Comment by psomers3** [comment]
Actually... seems like there is an issue where the weights get trained (loss goes down), but the weights that are saved are not these weights.

**Comment by philipperemy** [comment]
@psomers3 happy to hear that :)
Thanks a lot for your contributions.
Which PR should be merged first? is it this one or https://github.com/philipperemy/keras-tcn/pull/76 ?

**Comment by psomers3** [comment]
If the branches are to stay separate, it doesn’t really matter. But I think if we make sure #76 is good first, then it will definitely work here. tf2.0 is so much more flexible :D

**Comment by philipperemy** [comment]
Take some time to merge/rebase this branch with the latest master :)

**Comment by psomers3** [comment]
Will do. I think it would be a good idea to just add this to the pure keras and make them submodules. In other words have:
`import tcn.keras.TCN`
and 
`import tcn.tf.TCN` or `tcn.tensorflow.TCN`

**Comment by philipperemy** [comment]
@psomers3 I merge some TF2.0 code on master. Is this one still relevant now?

**Comment by psomers3** [comment]
No, this can be closed.

**Comment by philipperemy** [comment]
Thanks!
